---
title: "Sequencing_Report"
author: Finn
output:
  html_document:
    css: styles.css
    toc: yes
    number_section: yes
    fig_caption: true
  pdf_document:
    toc: yes
  always_allow_html: true
date: "`r format(Sys.time(), '%d %B %Y')`"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(fig.align = 'center', echo = TRUE)
```


``` {r libraries-and-data, echo = F, message = F, warning = F}
library(tidyverse)
library(data.table)
library(leaflet)
library(ggplot2)
library(RColorBrewer) #display.brewer.all()
library(colorRamps)
library(cowplot)
library(uwot)
library(leaflet.extras)
library(sp)
library(htmltools)
library(stringr)
library(patchwork)
library(readr)
library(knitr)
library(ggeffects)
library(scales)
library(tibble)
library(paletteer)

#library(parasiteLoad)
#library(pheatmap)
#library(dendextend)
#library(ctv)
#library(ape)
#library(phytools)



# Load Palette ####
r <- c(0, 64, 128, 179, 217, 255)
g <- c(0, 12, 25, 25, 12,  0)
b <- c(255, 249, 243, 191,  95,   0)
  
beach <- function (n, name = c("beach.colors")) 
  {
    beach.colors = rgb(r,g,b,maxColorValue = 255)
    name = match.arg(name)
    orig = eval(parse(text = name))
    rgb = t(col2rgb(orig))
    temp = matrix(NA, ncol = 3, nrow = n)
    x = seq(0, 1, , length(orig))
    xg = seq(0, 1, , n)
    for (k in 1:3) {
      hold = spline(x, rgb[, k], n = n)$y
      hold[hold < 0] = 0
      hold[hold > 255] = 255
      temp[, k] = round(hold)
    }
    palette = rgb(temp[, 1], temp[, 2], temp[, 3], maxColorValue = 255)
    palette
}

c <- c(245,  255,   192,   182)
d <- c(255,  184,   105,   255)
e <- c(91,    76,   105,    97)
#   yellow, orange, brown, green

Yl_Or_Gn <- function (n, name = c("Yl_Or_Gn")) 
  {
    Yl_Or_Gn = rgb(c,d,e,maxColorValue = 255)
    name = match.arg(name)
    orig = eval(parse(text = name))
    rgb = t(col2rgb(orig))
    temp = matrix(NA, ncol = 3, nrow = n)
    x = seq(0, 1, , length(orig))
    xg = seq(0, 1, , n)
    for (k in 1:3) {
      hold = spline(x, rgb[, k], n = n)$y
      hold[hold < 0] = 0
      hold[hold > 255] = 255
      temp[, k] = round(hold)
    }
    palette = rgb(temp[, 1], temp[, 2], temp[, 3], maxColorValue = 255)
    palette
}


SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") 
SOTA <- SOTA %>%
  mutate(Eim_Species = ifelse(eimeriaSpecies == "E_falciformis", "E_falciformis",
                              ifelse(eimeriaSpecies == "E_ferrisi", "E_ferrisi",
                                     ifelse(eimeriaSpecies == "Eimeria_alorani", "Eimeria_alorani",
                                            ifelse(eimeriaSpecies == "Eimeria_apionodes", "Eimeria_apionodes",
                                                   ifelse(eimeriaSpecies == "Eimeria_falciformis", "Eimeria_falciformis",
                                                          ifelse(eimeriaSpecies == "Eimeria_sp_Apodemus", "Eimeria_sp_Apodemus",
                                                                 ifelse(eimeriaSpecies == "Eimeria_vermiformis", "Eimeria_vermiformis",
                                                                        ifelse(eimeriaSpecies == "Negative", "Negative",
                                                                               ifelse(NA)))))))))) %>%
  mutate(Tested_for = case_when(ILWE_Crypto_Ct >= 0 & is.na(Ct.Eimeria) & is.na(Eim_Species) ~ "Crypto",
                                is.na(ILWE_Crypto_Ct) & !is.na(Eim_Species) ~ "Eimeria",
                                is.na(ILWE_Crypto_Ct) & (!is.na(Ct.Eimeria) | !is.na(Eim_Species)) ~ "Eimeria",
                                !is.na(ILWE_Crypto_Ct) & (!is.na(Ct.Eimeria) | !is.na(Eim_Species)) ~ "Crypto_Eimeria",
                                is.na(ILWE_Crypto_Ct) & is.na(Ct.Eimeria) & is.na(Eim_Species) ~ "none",
                                is.na(ILWE_Crypto_Ct) & (is.na(Ct.Eimeria) | is.na(Eim_Species)) ~ "none"),
         Crypto_Positive  = case_when(ILWE_Crypto_Ct > 0 ~ T,
                                      ILWE_Crypto_Ct == 0 ~ F),
         Eimeria_Positive = case_when(Eim_Species != "Negative" | Ct.Eimeria > 0 | OPG > 0 ~ T,
                                      Eim_Species == "Negative" | Ct.Eimeria == 0 | OPG == 0 ~ F),
         Infection = case_when(Tested_for == "Crypto" & ILWE_Crypto_Ct > 0                                                          ~ "Positive Crypto",
                               Tested_for == "Crypto" & ILWE_Crypto_Ct == 0                                                         ~ "Negative Crypto",
                               Tested_for == "Eimeria" & Eimeria_Positive == T                                               ~ "Positive Eimeria",
                               Tested_for == "Eimeria" & Eimeria_Positive == F                                               ~ "Negative Eimeria",
                               Tested_for == "Crypto_Eimeria" & ILWE_Crypto_Ct > 0  & (Ct.Eimeria > 0 |  Eim_Species != "Negative" | OPG > 0) ~ "Positive Crypto_Eimeria",
                               Tested_for == "Crypto_Eimeria" & ILWE_Crypto_Ct > 0  & (Ct.Eimeria == 0 | Eim_Species == "Negative"| OPG == 0) ~ "Positive Crypto",
                               Tested_for == "Crypto_Eimeria" & ILWE_Crypto_Ct == 0 & (Ct.Eimeria > 0 | Eim_Species != "Negative" | OPG > 0)  ~ "Positive Eimeria",
                               Tested_for == "Crypto_Eimeria" & ILWE_Crypto_Ct == 0 & (Ct.Eimeria == 0 | Eim_Species == "Negative" | OPG == 0) ~ "Negative Crypto_Eimeria",
                               Tested_for == "none" ~ "not tested"),
         Infected = case_when(Infection == "not tested" ~ F,
                              Infection == "Negative Crypto" ~ F,
                              Crypto_Positive == T | Eimeria_Positive == T ~ T,
                              Crypto_Positive == F & Eimeria_Positive == F ~ F,
                              is.na(Crypto_Positive) & Eimeria_Positive == F ~ F,
                              Crypto_Positive == F & is.na(Eimeria_Positive) ~ F,
                              ))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") 

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(Year == 2021, ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude))


Last_Sample <- SOTA[SOTA$Mouse_ID %like% "AA_", ] %>% select(Mouse_ID)
Last_Sample <- setDT(Last_Sample)[, paste0("Mouse_ID", 1:2) := tstrsplit(Mouse_ID, "_0")]


data_col_Eim_Infected = colorFactor(brewer.pal(3, "YlOrRd"), SOTA$Eimeria_Positive)
EIM_INFECTED <- SOTA %>% filter(Eimeria_Positive == T)

```


# 2021 Samples and previous Eimeria Infections

## Previous Eimeria Infections

- Crypto_Positives2021 ... positive Cases in 2021, color/radius == Ct
- Sample_Locations     ... Locations that have been sampled since 2016
- EIM_INFECTED         ... This location had samples with Eimeria Infection, radius == Infection intensity

```{r mini map Eimeria-Infections Past, echo = F, message = F, warning = F, include = FALSE}

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude), !is.na(HI))

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(Year == 2021, ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude))

SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") 

map <- Crypto_Detection_21 %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 52.6967 , lng =14.0833, zoom = 6)

# Cutting
Crypto_Detection$HI <- as.numeric(Crypto_Detection$HI)
Crypto_Detection$HI_Level <-  cut(Crypto_Detection$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$ILWE_Crypto_Ct, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))

data_col_HI       = colorFactor(beach(6), Crypto_Detection$HI)
data_col_HI_Level = colorFactor(beach(6), Crypto_Detection$HI_Level)
data_col_Ct_Level = colorFactor(brewer.pal(4, "YlOrRd"), Crypto_Detection_21$Ct_Level, reverse = T)
data_col_Sampled  = colorFactor(brewer.pal(6, "Accent"), SOTA$Year)

Ct_bl_30   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 30 & ILWE_Crypto_Ct > 0)
Ct_bl_34   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 34 & ILWE_Crypto_Ct > 30)
Ct_bl_36   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 36 & ILWE_Crypto_Ct > 34)
Ct_over_36 <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct > 36)
Crypto_Positives_21 <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct > 28)
Sampled <- SOTA %>% filter(!is.na(Year))




map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875  , 52.6053, 51.8978, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 13.5000), color = "purple", weight = 55, opacity = 0.1) %>%
  addCircleMarkers(data = Sampled,
                   col = "#351c75",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 0.3,
                   radius = 3,
                   group = "Sample Locations") %>%
  addCircleMarkers(data = EIM_INFECTED,
                   col = "#72cac3",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 0.3,
                   radius = sqrt((EIM_INFECTED$delta_ct_cewe_MminusE)^2),
                   group = "EIM_INFECTED") %>%
  addCircleMarkers(data = Crypto_Positives_21, 
                   color = ~data_col_Ct_Level(Ct_Level),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>", as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>Crypto Infections:<b>",  as.character(Crypto_mus_caught), "<b>/<b>", as.character(mus_caught), "<br>",
                                  "<b>Oocyst Prediction:<b>",  as.character(Oocyst_Predict_Crypto), "<br>",
                                  sep=" "),
                   radius = (Crypto_Positives_21$Oocyst_Predict_Crypto)/1000,
                   #opacity = (Crypto_Positives_21$ILWE_Crypto_Ct),
                   opacity = (Crypto_Positives_21$Oocyst_Predict_Crypto)/1000,
                   group = "Crypto Positives 2021") %>%
  addLayersControl(overlayGroups = c("Sample Locations", "EIM_INFECTED"),
                   baseGroups = c("Crypto Positives 2021"),
                 options = layersControlOptions(collapsed = F)) %>%
  addLegend("bottomleft", 
            pal = data_col_Ct_Level, 
            title = "Ct",
            values = Crypto_Detection_21$Ct_Level, 
            group = c("Crypto Positives 2021"),
            opacity = 1)
```

## Highest Crypto-Infections in 2021
``` {r Table of Infected Samples, message = F, warning = F, echo = F}

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(Year == 2021, ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude))


Crypto_Detection_21 <- Crypto_Detection_21 %>% 
  filter(ILWE_Crypto_Ct > 0) %>% 
  select(Mouse_ID, ILWE_Crypto_Ct, Oocyst_Predict_Crypto, Latitude, Longitude) %>% arrange(desc(Oocyst_Predict_Crypto))

kable(Crypto_Detection_21[1:12,],
      col.names = c("Mouse_ID", "ILWE_Crypto_Ct", "Oocyst_Predict_Crypto", "Latitude", "Longitude"),
      #align = "ccc",
      caption = "Table 1: _Cryptosporidium_-positive samples from 2021, arranged by ILWE_Crypto_Ct")
```

## Testing for Eimeria in Crypto-infected Samples

- Green arrows indicate samples that I sent
- Samples AA_0901 and AA_0902 showed (strong) amplification for the COI Eimeria marker and where therefore not among the samples that I sent you






# Cryptosporidium Detection in Wild Samples in the House Mouse Hybrid Zone
Since 2016, `r max(Last_Sample$Mouse_ID2)` mice were sampled by this workgroup, `r Crypto_Detection %>% filter(ILWE_Crypto_Ct >= 0) %>% count()` were tested for Cryptosporidium sp. by performing qPCR on DNA extracted from Ileum wall tissue.
Out of these `r Crypto_Detection %>% filter(ILWE_Crypto_Ct >= 0) %>% count()` samples, `r Crypto_Detection %>% filter(ILWE_Crypto_Ct > 0) %>% count() %>% sum()` tested positive for Crypto. An average Cryptosporidium prevalence of `r round((Crypto_Detection %>% filter(ILWE_Crypto_Ct > 0) %>% count() / Crypto_Detection %>% filter(ILWE_Crypto_Ct >= 0) %>% count()) * 100, digits = 2)`% was detected in the House Mouse Hybrid Zone.

``` {r Pos-Samples-Yr, echo = F}
Pos_Samples_Yr <- SOTA %>% filter(ILWE_Crypto_Ct > 0, Year >= 2016) %>% group_by(Year) %>% count()
Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")

All_Samples_Yr <- Crypto_Detection %>% group_by(Year) %>% count()
Samples_Yr <- full_join(Pos_Samples_Yr, All_Samples_Yr, by = "Year") %>% mutate(Prevalence = (n.x / n.y)) %>% filter(!is.na(Prevalence))

Samples_Yr %>%
  ggplot(aes(x = Year, label = Prevalence)) +
      geom_col(aes(y = n.y, fill = "blue")) +
      geom_col(aes(y = n.x, fill = "red")) +
      geom_text(aes(label = percent(Prevalence),
                    y = (n.x / n.y)), nudge_y = 9) +
      labs(y = "Samples [n]") +
  theme(legend.position = "none") +
  ggtitle("Cryptosporidium spp. prevalence in the HMHZ since 2016")
  
```

## Table of Highly Infected Samples
``` {r Top-infected, echo = F}
Samples <- Crypto_Detection %>% select(Mouse_ID, ILWE_Crypto_Ct, Oocyst_Predict_Crypto, Year, Latitude, Longitude) %>% filter(ILWE_Crypto_Ct > 0) %>% arrange(ILWE_Crypto_Ct) %>% head(20)

kable(Samples, caption = "Table 1.1: Crypto-Positive Samples collected since 2016, arranged by ILWE_Crypto_Ct, 20 highest infected samples displayed", align = c("cccccc"))

Samples <- Samples[colnames(a) %in% c("Mouse_ID", "ILWE_Crypto_Ct", "Oocyst_Predict_Crypto", "Latitude", "Longitude")] 
write.csv(Samples, "highest_infected_samples.csv")
```

## Geographic distribution of positive Samples in the HMHZ

```{r mini map HI , echo = F, message = F, warning = F}
SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv")  %>% filter(!is.na(Latitude), !is.na(Longitude), !is.na(HI), !Mouse_ID %in% c("SK_2697", "SK_3173", "SK_905"))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") 

Crypto_Detection_tested <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(ILWE_Crypto_Ct >= 0, !is.na(Latitude), !is.na(Longitude))

Crypto_Positive <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude))


Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")  %>% filter(Year == 2021, ILWE_Crypto_Ct > 0, !is.na(Latitude), !is.na(Longitude))

High_Infection_Samples <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_input/Cryptosporidium/High_Infection_Samples.csv")

Western_Hybrid <- High_Infection_Samples %>% filter(HI >= 0.5)
Eastern_Hybrid <- High_Infection_Samples %>% filter(HI < 0.50)


map <- Crypto_Detection_21 %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 52.520007, lng =13.404954, zoom = 7)

# Cutting
SOTA$HI <- as.numeric(SOTA$HI)
SOTA$HI_Level <-  cut(SOTA$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$ILWE_Crypto_Ct, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))


SeqSamples <- Crypto_Detection %>%
  mutate(Sequenced = Mouse_ID %in% c("AA_0144", "AA_0325", "AA_0689", "AA_0209",
                                     "AA_0282", "AA_0793", "AA_0667", "AA_0805", 
                                     "AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585",
                                     "AA_0546", "AA_0589", "AA_0571", "AA_0667"))


Ct_bl_30   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 30 & ILWE_Crypto_Ct > 0)
Ct_bl_34   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 34 & ILWE_Crypto_Ct > 30)
Ct_bl_36   <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct <= 36 & ILWE_Crypto_Ct > 34)
Ct_over_36 <- Crypto_Detection_21 %>% filter(ILWE_Crypto_Ct > 36)
Sequenced  <- SeqSamples %>% filter(Sequenced == T)
Not_Sequenced <- SeqSamples %>% filter(Sequenced == F)

data_col_HI       = colorFactor(beach(6), SOTA$HI)
data_col_HI_Level = colorFactor(beach(6), SOTA$HI_Level)
data_col_Ct_Level = colorFactor(brewer.pal(4, "YlOrRd"), Crypto_Detection_21$Ct_Level, reverse = T)
data_col_Sequenced = colorFactor(beach(2), SeqSamples$HI)


map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875  , 52.6053, 51.8978, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 13.5000), color = "purple", weight = 55, opacity = 0.1) %>%
  addCircleMarkers(data = SOTA,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  #"<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  #"<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (total)") %>%
  addCircleMarkers(data = Crypto_Detection_tested,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  #"<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  #"<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (Crypto-tested)") %>%
  addCircleMarkers(data = Crypto_Positive,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  #"<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  #"<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (Crypto-positive)") %>%
  addCircleMarkers(data = Sequenced, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  #"<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  #"<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Sequenced") %>%
  addCircleMarkers(data = High_Infection_Samples, 
                   color = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  #"<b>Ct:<b>",      as.character(ILWE_Crypto_Ct),"<br>",
                                  #"<b>Oocysts:<b>", as.character(Oocyst_Predict_Crypto), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 3,
                   radius = 3,
                   group = "High_Infection_Samples") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'),
            opacity = 1) %>%
addLayersControl(baseGroups = c("Samples (total)", "Samples (Crypto-tested)", "Samples (Crypto-positive)", "High_Infection_Samples", "Sequenced"), 
                 options = layersControlOptions(collapsed = F))
```

# Genes of Interest

## Genes and Gene_IDs
``` {r Gene-IDs, echo = F, message = F, warning = F}
Gene_IDs <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Gene_IDs.csv")


kable(Gene_IDs, align = c("lccccccccc"))
```


# Sequencing Results from Kvac study and EH study

``` {r Map Prep, echo = F, message = F, warning = F}
SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv")  %>% filter(!is.na(Latitude), !is.na(Longitude), !is.na(HI), !Mouse_ID %in% c("SK_2697", "SK_3173", "SK_905"))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") 

High_Infection_Samples <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_input/Cryptosporidium/High_Infection_Samples.csv")

HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))
HMHZ$HI[ HMHZ$Mouse_ID %in% "AA_0900"] <- 0.94444444


map <- HMHZ %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 50.07245, lng =12.92014, zoom = 6)

# Cutting
SOTA$HI <- as.numeric(SOTA$HI)
SOTA$HI_Level <-  cut(SOTA$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$ILWE_Crypto_Ct, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))



GP60     <- HMHZ %>% filter(!is.na(GP60_Subtype))
Actin    <- HMHZ %>% filter(Actin_Subtype %in% c("A1", "A2"))
TRAP_C1  <- HMHZ %>% filter(TRAP_C1_Subtype %in% c("T1", "T2"))
COWP     <- HMHZ %>% filter(COWP_Subtype %in% c("C1", "C2"))


data_col_HI    = colorFactor(beach(6), HMHZ$HI)
data_col_GP60  = colorFactor(beach(9), GP60$GP60_Subtype, rev = T)
data_col_Actin = colorFactor(beach(2), Actin$Actin_Subtype, rev = T)
data_col_Trap  = colorFactor(beach(2), TRAP_C1$TRAP_C1_Subtype, rev = T)
data_col_COWP  = colorFactor(beach(2), COWP$COWP_Subtype, rev = T)


SeqSamples <- Crypto_Detection %>%
  mutate(Sequenced = Mouse_ID %in% c("AA_0144", "AA_0325", "AA_0689", "AA_0209",
                                     "AA_0282", "AA_0793", "AA_0667", "AA_0805", 
                                     "AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585",
                                     "AA_0546", "AA_0589", "AA_0571"),
         TRAP_C1_Subtype = case_when(Mouse_ID %in% c("AA_0546", "AA_0578", "AA_0580", "AA_0589") ~ "Eastern",
                                     Mouse_ID %in% c("AA_0571", "AA_0667") ~ "Western"),
         GP60_Subtype = case_when(Mouse_ID %in% c("AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585") ~ "IXb"),
         MSC6_Subtype = case_when(Mouse_ID %in% c("AA_0282", "AA_0523") ~ "Substitution T->A @236nt and C->T @350",
                                  Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0667", "AA_0689", "AA_0793", "AA_0805", "AA_0900") ~ "A @236nt and T @350"),
         GST_Subtype = case_when(Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "A->T @519",
                         Mouse_ID %in% c("AA_0523") ~ "like RefSeq"),
         SKSR_Subtype = case_when(Mouse_ID %in% c("AA_0523") ~ "more SNPs",
                                  Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "different from AA_0523"),
         MEDLE_Subtype = case_when(Mouse_ID %in% c("AA_0523") ~ "Deletion of 9nt @446",
                                   Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "Deletion of 6nt @446"))

```


## Kvac Study Results
``` {r Kvac-Samples-Table, echo = F, message = F, fig.align = 'left'}
HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))
HMHZ$HI[ HMHZ$Mouse_ID %in% "AA_0900"] <- 0.94444444

Kvac <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Kvac_Samples_Locations1.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude)) 

colnames(Kvac)[colnames(Kvac)%in%"TRAP.C1_Subtype"] <- "TRAP_C1_Subtype"

Kvac <- left_join(Kvac, HMHZ)  %>% select(1, 7:11)  %>% mutate(GP60_Subtype = case_when(!is.na(GP60) ~ "seq")) %>% select(-6)

Kvac <- Kvac %>% 
  pivot_longer(cols = c(2:5), names_to = "Marker", values_to = "Sequenced1") %>% 
  mutate(Sequenced = case_when(!is.na(Sequenced1) ~ TRUE,
                               is.na(Sequenced1)  ~ FALSE)) %>% 
  select(-Sequenced1)

Kvac %>% 
  ggplot(aes(x = Marker, y = Mouse_ID, fill = Sequenced)) +
  geom_tile() +
  scale_fill_paletteer_d("beyonce::X2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

## HMHZ Sequencing Reults
``` {r HMHZ-Samples-Table, echo = F, message = F, fig.align = 'left'}
HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))
HMHZ$HI[ HMHZ$Mouse_ID %in% "AA_0900"] <- 0.94444444

HMHZ <- HMHZ[HMHZ$Mouse_ID %like% "AA_", ]

HMHZ[,c(1:6, 8:17)] %>% kable(align = c("ccccccccccccccccc"))

```


## Map of Sequencing Results
```{r mini map Sequencing-Results, echo = F, message = F, warning = F}
HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))
HMHZ$HI[ HMHZ$Mouse_ID %in% "AA_0900"] <- 0.94444444

HMHZ$HI_Level <-  cut(HMHZ$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'))

HMHZ <- HMHZ %>%
  mutate(Species = ifelse(Mouse_ID %in% c('AA_0325', 'AA_0669'), 'other',
                          ifelse(NA)),
         Species = case_when(is.na(Species) ~ 'C.tyzzeri',
                             Mouse_ID == 'AA_0325' ~ 'C.parvum', 
                             Mouse_ID == 'AA_0669' ~ 'C.meleagridis'),
         GP60_Ssp = case_when(Mouse_ID %in% c("CR_2085", "CR_2084", "CR_2090",
                                              "CR_2125", "CR_2126", "CR_2127",
                                              "CR_2128", "CR_2149", "CR_2208", "CR_2206") ~ "IXa",
                              Mouse_ID %in% c("AA_0523", "AA_0282") ~ "IXb.1",
                              Mouse_ID %in% c("G_2136", "CR_2163", "G_2099",
                                              "CR_2147", "G_2135", "G_2177",
                                              "G_2179", "G_2174", "AA_0534",
                                              "G_2103", "AA_0537", "AA_0585",
                                              "CR_4293", "G_2116", "G_2120",
                                              "G_2134", "G_2160", "G_2108",
                                              "G_2169", "G_3224") ~ "IXb.2",
                              Mouse_ID %in% c("AA_0209", "AA_0144", "AA_0325", "AA_0545", "AA_0553",
                                              "AA_0554", "AA_0580",  "AA_0667", "AA_0689", "AA_0793",
                                              "AA_0805","AA_0900", "NZ_1633", "NZ_1634", "NZ_1635",
                                              "NZ_1636", "NZ_1639", "NZ_1640", "NZ_1641",
                                              "NZ_1642", "NZ_1644", "Tyz-GP60") ~ "IXb.3"))

GST   <- HMHZ %>% filter(!is.na(GST_Subtype))
MSC6  <- HMHZ %>% filter(!is.na(MSC6_Subtype))
GP60  <- HMHZ %>% filter(!is.na(GP60_Subtype))
TRAP_C1  <- HMHZ %>% filter(!is.na(TRAP_C1_Subtype))
SKSR  <- HMHZ %>% filter(!is.na(SKSR_Subtype))
MEDLE <- HMHZ %>% filter(!is.na(MEDLE_Subtype))
CP56  <- HMHZ %>% filter(!is.na(CP56_Subtype))
COWP  <- HMHZ %>% filter(!is.na(COWP_Subtype))
GP60_Sub  <- HMHZ %>% filter(!is.na(GP60_Ssp))
Species   <- HMHZ %>% filter(!is.na(Species))

data_col_HI           = colorFactor(     beach(6), HMHZ$HI)
data_col_HI_Level     = colorFactor(     beach(6), HMHZ$HI_Level)
data_col_HMHZ         = colorFactor(     beach(2), HMHZ$HI)
data_col_MSC6         = colorFactor(  Yl_Or_Gn(3), HMHZ$MSC6_Subtype,    reverse = F)
data_col_GP60         = colorFactor(  Yl_Or_Gn(3), HMHZ$GP60_Subtype,    reverse = T)
data_col_GP60_Ssp     = colorFactor(  Yl_Or_Gn(4), HMHZ$GP60_Ssp,        reverse = T)
data_col_GST          = colorFactor(  Yl_Or_Gn(3), HMHZ$GST_Subtype,     reverse = F)
data_col_Trap         = colorFactor(  Yl_Or_Gn(3), HMHZ$TRAP_C1_Subtype, reverse = F)
data_col_SKSR         = colorFactor(  Yl_Or_Gn(3), HMHZ$SKSR_Subtype,    reverse = F)
data_col_MEDLE        = colorFactor(  Yl_Or_Gn(3), HMHZ$MEDLE_Subtype,   reverse = F)
data_col_CP56         = colorFactor(  Yl_Or_Gn(3), HMHZ$CP56_Subtype,    reverse = F)
data_col_Actin        = colorFactor(  Yl_Or_Gn(3), HMHZ$Actin_Subtype,   reverse = F)
data_col_COWP         = colorFactor(  Yl_Or_Gn(3), HMHZ$COWP_Subtype,    reverse = F)
data_col_Species      = colorFactor(brewer.pal(3, "Set3"), HMHZ$Species, reverse = F)

#display.brewer.all()


map <- Sequenced %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 51.676761 , lng =13.679507, zoom = 6)

map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875, 52.6053, 51.8978, 50.08506775 , 48.13659738768638, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 12.5988909, 11.582597486663213, 11.50000), color = "purple", weight = 75, opacity = 0.1)  %>%
  addCircleMarkers(data = HMHZ, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 0.5,
                   radius = 6,
                   group = "Colored by HI") %>%
  addCircleMarkers(data = HMHZ, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 0.5,
                   radius = 3,
                   group = "Colored by HI_small_rad") %>%
  addCircleMarkers(data = Species, 
                   col = ~data_col_Species(Species),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>Species:<b>",      as.character(Species), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 2,
                   group = "Colored by Species") %>%
    addCircleMarkers(data = MSC6, 
                   col = ~data_col_MSC6(MSC6_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>", 
                                  "<b>GP60 Subtype:<b>",        as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by MSC6-7") %>%
  
  addCircleMarkers(data = GST, 
                   col = ~data_col_GST(GST_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GST") %>%
  addCircleMarkers(data = SKSR, 
                   col = ~data_col_SKSR(SKSR_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by SKSR") %>%
  addCircleMarkers(data = MEDLE, 
                   col = ~data_col_MEDLE(MEDLE_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by MEDLE") %>%
  addCircleMarkers(data = CP56, 
                   col = ~data_col_CP56(CP56_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by CP56") %>%
  addCircleMarkers(data  = GP60, 
                   col   = ~data_col_GP60(GP60_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  "<b>GP60 Detail Subtype:<b>", as.character(GP60_Ssp), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GP60") %>%
  addCircleMarkers(data  = GP60_Sub, 
                   col   = ~data_col_GP60_Ssp(GP60_Ssp),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  "<b>GP60 Detail Subtype:<b>", as.character(GP60_Ssp), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GP60 Subtype Family") %>%
  addCircleMarkers(data = TRAP_C1, 
                   col = ~data_col_Trap(TRAP_C1_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3, 
                   group = "Colored by TRAP-C1") %>%
    addCircleMarkers(data  = Actin, 
                   col   = ~data_col_Actin(Actin_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude),  "<br>",
                                  "<b>GP60 Subtype:<b>",       as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  
                                  sep=" "),
                   opacity = 5,
                   radius  = 3,
                   group   = "Colored by Actin") %>%
  addCircleMarkers(data  = COWP, 
                   col   = ~data_col_COWP(COWP_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude),
                                  "<b>GP60 Subtype:<b>",       as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius  = 3,
                   group   = "Colored by COWP") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'),
            opacity = 1) %>%
  addLegend("bottomright", 
            pal = data_col_Species, 
            title = "Species",
            values = HMHZ$Species, 
            group = c('C.tyzzeri', 'C.parvum', 'C.meleagridis'),
            opacity = 1) %>%
addLayersControl(baseGroups = c("Colored by HI",
                                "Colored by HI_small_rad",
                                "Colored by MSC6-7", 
                                "Colored by GST",
                                "Colored by SKSR",
                                "Colored by MEDLE",
                                "Colored by CP56",
                                "Colored by GP60",
                                "Colored by GP60 Subtype Family",
                                "Colored by TRAP-C1",
                                "Colored by COWP",
                                "Colored by Actin"),
                 overlayGroups = c("Colored by HI",
                                  "Colored by Species"), 
                 options = layersControlOptions(collapsed = T))


```

## Map of Clade Memberships

### MSC6-7
```{r mini map Clades, echo = F, message = F, warning = F}
#![Population Structure HMHZ](https://raw.githubusercontent.com/tlobnow/Crypto/main/Map_BSc.jpeg)

HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))
HMHZ$HI[ HMHZ$Mouse_ID %in% "AA_0900"] <- 0.94444444

Clades <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Clade_Memberships.csv")

#Sequenced <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv")

Clades$HI_Level <-  cut(Clades$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'))

GST   <- Clades %>% filter(!is.na(GST))


TRAP_C1  <- Clades %>% filter(!is.na(TRAP_C1))
SKSR  <- Clades %>% filter(!is.na(SKSR))
MEDLE <- Clades %>% filter(!is.na(MEDLE))
CP56  <- Clades %>% filter(!is.na(CP56))

Actin  <- Clades %>% filter(!is.na(Actin))

Light_green <- Clades %>% filter(GP60    == "IXa"|
                                 Actin   == "A1"|
                                 COWP    == "C1"|
                                 TRAP_C1 == "T1")

Yellow <- Clades %>% filter(GP60    == "IXb.2"|
                            Actin   == "A2"|
                            COWP    == "C2"|
                            TRAP_C1 == "T2"|
                            MSC6    == "MS1"|
                            GST     == "G1"|
                            CP56    == "CP1"|
                            SKSR    == "S1"|
                            MEDLE   == "ME1")
Orange <- Clades %>% filter(GP60    == "IXb.1"|
                            Actin   == "A3"|
                            MSC6    == "MS2"|
                            GST     == "G2"|
                            CP56    == "CP2")
Dark_Orange <- Clades %>% filter(GP60 == "IXb.3")




data_col_HI           = colorFactor(beach(6), Clades$HI)
data_col_HI_Level     = colorFactor(beach(6), Clades$HI_Level)
data_col_Clades       = colorFactor(beach(2), Clades$HI)



MSC6  <- Clades %>% filter(!is.na(MSC6))
  MS1 <- MSC6 %>% filter(MSC6 == "MS1")
  MS2 <- MSC6 %>% filter(MSC6 == "MS2")

map <- Clades %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 51.676761 , lng =13.679507, zoom = 6)

map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875, 52.6053, 51.8978, 50.08506775 , 48.13659738768638, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 12.5988909, 11.582597486663213, 11.50000), color = "purple", weight = 75, opacity = 0.1)  %>%
  addCircleMarkers(data = Clades, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 0.1,
                   radius = 10,
                   group = "Colored by HI") %>%
  addCircleMarkers(data = MS1, 
                   col = "yellow",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "MS1") %>%
  addCircleMarkers(data = MS2, 
                   col = "orange",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "MS2") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'),
            opacity = 1)

```

### COWP
```{r, echo = F, message = F, warning = F}
COWP  <- Clades %>% filter(!is.na(COWP))
  C1 <- COWP %>% filter(COWP == "C1")
  C2 <- COWP %>% filter(COWP == "C2")

map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875, 52.6053, 51.8978, 50.08506775 , 48.13659738768638, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 12.5988909, 11.582597486663213, 11.50000), color = "purple", weight = 75, opacity = 0.1)  %>%
  addCircleMarkers(data = Clades, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 0.1,
                   radius = 10,
                   group = "Colored by HI") %>%
  addCircleMarkers(data = C1, 
                   col = "#8fce00",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "C1") %>%
  addCircleMarkers(data = C2, 
                   col = "orange",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "C2") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'),
            opacity = 1)

```

### GP60
```{r, echo = F, message = F, warning = F}
GP60  <- Clades %>% filter(!is.na(GP60))
  IXa <- GP60 %>% filter(GP60 == "IXa")
  IXb.1 <- GP60 %>% filter(GP60 == "IXb.1")
  IXb.2 <- GP60 %>% filter(GP60 == "IXb.2")
  IXb.3 <- GP60 %>% filter(GP60 == "IXb.3")


map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875, 52.6053, 51.8978, 50.08506775 , 48.13659738768638, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 12.5988909, 11.582597486663213, 11.50000), color = "purple", weight = 75, opacity = 0.1)  %>%
  addCircleMarkers(data = Clades, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 0.1,
                   radius = 10,
                   group = "Colored by HI") %>%
  addCircleMarkers(data = IXa, 
                   col = "#8fce00",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "IXa") %>%
  addCircleMarkers(data = IXb.1, 
                   col = "#fbe419",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "IXb.1") %>%
  addCircleMarkers(data = IXb.2, 
                   col = "orange",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "IXb.2") %>%
  addCircleMarkers(data = IXb.3, 
                   col = "#b46d23",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "IXb.3") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0.00', 'HI < 0.25', 'HI < 0.50', 'HI < 0.75', 'HI < 1.00', 'HI = 1.00'),
            opacity = 1)

```
