---
title: "Sequencing_Report"
author: Finn
output:
  html_document:
    css: styles.css
    toc: yes
    number_section: yes
    fig_caption: true
  pdf_document:
    toc: yes
  always_allow_html: true
date: "`r format(Sys.time(), '%d %B %Y')`"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(fig.align = 'center', echo = TRUE)
```


``` {r libraries-and-data, echo = F, message = F}
library(tidyverse)
library(data.table)
library(leaflet)
library(ggplot2)
library(RColorBrewer) #display.brewer.all()
library(colorRamps)
library(cowplot)
library(uwot)
library(leaflet.extras)
library(sp)
library(htmltools)
library(stringr)
library(patchwork)
library(readr)
library(knitr)
library(ggeffects)
library(parasiteLoad)
library(scales)
library(pheatmap)
library(dendextend)
library(paletteer)


# Load Palette ####
r <- c(0, 64, 128, 179, 217, 255)
g <- c(0, 12, 25, 25, 12,  0)
b <- c(255, 249, 243, 191,  95,   0)
  
beach <- function (n, name = c("beach.colors")) 
  {
    beach.colors = rgb(r,g,b,maxColorValue = 255)
    name = match.arg(name)
    orig = eval(parse(text = name))
    rgb = t(col2rgb(orig))
    temp = matrix(NA, ncol = 3, nrow = n)
    x = seq(0, 1, , length(orig))
    xg = seq(0, 1, , n)
    for (k in 1:3) {
      hold = spline(x, rgb[, k], n = n)$y
      hold[hold < 0] = 0
      hold[hold > 255] = 255
      temp[, k] = round(hold)
    }
    palette = rgb(temp[, 1], temp[, 2], temp[, 3], maxColorValue = 255)
    palette
  }

SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") %>% select(-X)
SOTA <- SOTA %>%
  mutate(Eim_Species = ifelse(eimeriaSpecies == "E_falciformis", "E_falciformis",
                              ifelse(eimeriaSpecies == "E_ferrisi", "E_ferrisi",
                                     ifelse(eimeriaSpecies == "Eimeria_alorani", "Eimeria_alorani",
                                            ifelse(eimeriaSpecies == "Eimeria_apionodes", "Eimeria_apionodes",
                                                   ifelse(eimeriaSpecies == "Eimeria_falciformis", "Eimeria_falciformis",
                                                          ifelse(eimeriaSpecies == "Eimeria_sp_Apodemus", "Eimeria_sp_Apodemus",
                                                                 ifelse(eimeriaSpecies == "Eimeria_vermiformis", "Eimeria_vermiformis",
                                                                        ifelse(eimeriaSpecies == "Negative", "Negative",
                                                                               ifelse(NA)))))))))) %>%
  mutate(Tested_for = case_when(Ct_mean >= 0 & is.na(Ct.Eimeria) & is.na(Eim_Species) ~ "Crypto",
                                is.na(Ct_mean) & !is.na(Eim_Species) ~ "Eimeria",
                                is.na(Ct_mean) & (!is.na(Ct.Eimeria) | !is.na(Eim_Species)) ~ "Eimeria",
                                !is.na(Ct_mean) & (!is.na(Ct.Eimeria) | !is.na(Eim_Species)) ~ "Crypto_Eimeria",
                                is.na(Ct_mean) & is.na(Ct.Eimeria) & is.na(Eim_Species) ~ "none",
                                is.na(Ct_mean) & (is.na(Ct.Eimeria) | is.na(Eim_Species)) ~ "none"),
         Crypto_Positive  = case_when(Ct_mean > 0 ~ T,
                                      Ct_mean == 0 ~ F),
         Eimeria_Positive = case_when(Eim_Species != "Negative" | Ct.Eimeria > 0 | OPG > 0 ~ T,
                                      Eim_Species == "Negative" | Ct.Eimeria == 0 | OPG == 0 ~ F),
         Infection = case_when(Tested_for == "Crypto" & Ct_mean > 0                                                          ~ "Positive Crypto",
                               Tested_for == "Crypto" & Ct_mean == 0                                                         ~ "Negative Crypto",
                               Tested_for == "Eimeria" & Eimeria_Positive == T                                               ~ "Positive Eimeria",
                               Tested_for == "Eimeria" & Eimeria_Positive == F                                               ~ "Negative Eimeria",
                               Tested_for == "Crypto_Eimeria" & Ct_mean > 0  & (Ct.Eimeria > 0 |  Eim_Species != "Negative" | OPG > 0) ~ "Positive Crypto_Eimeria",
                               Tested_for == "Crypto_Eimeria" & Ct_mean > 0  & (Ct.Eimeria == 0 | Eim_Species == "Negative"| OPG == 0) ~ "Positive Crypto",
                               Tested_for == "Crypto_Eimeria" & Ct_mean == 0 & (Ct.Eimeria > 0 | Eim_Species != "Negative" | OPG > 0)  ~ "Positive Eimeria",
                               Tested_for == "Crypto_Eimeria" & Ct_mean == 0 & (Ct.Eimeria == 0 | Eim_Species == "Negative" | OPG == 0) ~ "Negative Crypto_Eimeria",
                               Tested_for == "none" ~ "not tested"),
         Infected = case_when(Infection == "not tested" ~ F,
                              Infection == "Negative Crypto" ~ F,
                              Crypto_Positive == T | Eimeria_Positive == T ~ T,
                              Crypto_Positive == F & Eimeria_Positive == F ~ F,
                              is.na(Crypto_Positive) & Eimeria_Positive == F ~ F,
                              Crypto_Positive == F & is.na(Eimeria_Positive) ~ F,
                              ))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X)

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Year == 2021, Ct_mean > 0, !is.na(Latitude), !is.na(Longitude))


Last_Sample <- SOTA[SOTA$Mouse_ID %like% "AA_", ] %>% select(Mouse_ID)
Last_Sample <- setDT(Last_Sample)[, paste0("Mouse_ID", 1:2) := tstrsplit(Mouse_ID, "_0")]


data_col_Eim_Infected = colorFactor(brewer.pal(3, "YlOrRd"), SOTA$Eimeria_Positive)
EIM_INFECTED <- SOTA %>% filter(Eimeria_Positive == T)

```


# 2021 Samples and previous Eimeria Infections

## Previous Eimeria Infections

- Crypto_Positives2021 ... positive Cases in 2021, color/radius == Ct
- Sample_Locations     ... Locations that have been sampled since 2016
- EIM_INFECTED         ... This location had samples with Eimeria Infection, radius == Infection intensity

```{r mini map Eimeria-Infections Past, echo = F, message = F, warning = F, include = TRUE}

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Ct_mean > 0, !is.na(Latitude), !is.na(Longitude), !is.na(HI))

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Year == 2021, Ct_mean > 0, !is.na(Latitude), !is.na(Longitude))

SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") %>% select(-X)

map <- Crypto_Detection_21 %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 52.6967 , lng =14.0833, zoom = 10)

# Cutting
Crypto_Detection$HI <- as.numeric(Crypto_Detection$HI)
Crypto_Detection$HI_Level <-  cut(Crypto_Detection$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$Ct_mean, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))

data_col_HI       = colorFactor(beach(6), Crypto_Detection$HI)
data_col_HI_Level = colorFactor(beach(6), Crypto_Detection$HI_Level)
data_col_Ct_Level = colorFactor(brewer.pal(4, "YlOrRd"), Crypto_Detection_21$Ct_Level, reverse = T)
data_col_Sampled  = colorFactor(brewer.pal(6, "Accent"), SOTA$Year)

Ct_bl_30   <- Crypto_Detection_21 %>% filter(Ct_mean <= 30 & Ct_mean > 0)
Ct_bl_34   <- Crypto_Detection_21 %>% filter(Ct_mean <= 34 & Ct_mean > 30)
Ct_bl_36   <- Crypto_Detection_21 %>% filter(Ct_mean <= 36 & Ct_mean > 34)
Ct_over_36 <- Crypto_Detection_21 %>% filter(Ct_mean > 36)
Crypto_Positives_21 <- Crypto_Detection_21 %>% filter(Ct_mean > 28)
Sampled <- SOTA %>% filter(!is.na(Year), Year != 2021)




map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875  , 52.6053, 51.8978, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 13.5000), color = "purple", weight = 55, opacity = 0.1) %>%
  addCircleMarkers(data = Sampled,
                   col = "#351c75",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 0.3,
                   radius = 3,
                   group = "Sample Locations") %>%
  addCircleMarkers(data = EIM_INFECTED,
                   col = "#72cac3",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 0.3,
                   radius = sqrt((EIM_INFECTED$delta_ct_cewe_MminusE)^2),
                   group = "EIM_INFECTED") %>%
  addCircleMarkers(data = Crypto_Positives_21, 
                   color = ~data_col_Ct_Level(Ct_Level),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>", as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>Crypto Infections:<b>",  as.character(Crypto_mus_caught), "<b>/<b>", as.character(mus_caught), "<br>",
                                  "<b>Oocyst Prediction:<b>",  as.character(Oocyst_Predict), "<br>",
                                  sep=" "),
                   radius = (Crypto_Positives_21$Oocyst_Predict)/1000,
                   #opacity = (Crypto_Positives_21$Ct_mean),
                   opacity = (Crypto_Positives_21$Oocyst_Predict)/1000,
                   group = "Crypto Positives 2021") %>%
  addLayersControl(overlayGroups = c("Sample Locations", "EIM_INFECTED"),
                   baseGroups = c("Crypto Positives 2021"),
                 options = layersControlOptions(collapsed = F)) %>%
  addLegend("bottomleft", 
            pal = data_col_Ct_Level, 
            title = "Ct",
            values = Crypto_Detection_21$Ct_Level, 
            group = c("Crypto Positives 2021"),
            opacity = 1)
```

## Highest Crypto-Infections in 2021
``` {r Table of Infected Samples, message = F, warning = F, echo = F}

Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Year == 2021, Ct_mean > 0, !is.na(Latitude), !is.na(Longitude))

Crypto_Detection_21 <- Crypto_Detection_21 %>% 
  filter(Ct_mean > 0) %>% 
  select(Mouse_ID, Ct_mean, Oocyst_Predict, Latitude, Longitude) %>% arrange(desc(Oocyst_Predict))

kable(Crypto_Detection_21[1:12,],
      col.names = c("Mouse_ID", "Ct_mean", "Oocyst_Predict", "Latitude", "Longitude"),
      #align = "ccc",
      caption = "Table 1: _Cryptosporidium_-positive samples from 2021, arranged by Ct_mean")
```

## Testing for Eimeria in Crypto-infected Samples

- Green arrows indicate samples that I sent
- Samples AA_0901 and AA_0902 showed (strong) amplification for the COI Eimeria marker and where therefore not among the samples that I sent you

![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/PCRs/Screenshot%202021-10-26%20at%2010.20.44%20AM.png)






# Cryptosporidium Detection in Wild Samples in the House Mouse Hybrid Zone
Since 2016, `r max(Last_Sample$Mouse_ID2)` mice were sampled by this workgroup, `r Crypto_Detection %>% filter(Ct_mean >= 0) %>% count()` were tested for Cryptosporidium sp. by performing qPCR on DNA extracted from Ileum wall tissue.
Out of these `r Crypto_Detection %>% filter(Ct_mean >= 0) %>% count()` samples, `r Crypto_Detection %>% filter(Ct_mean > 0) %>% count() %>% sum()` tested positive for Crypto. An average Cryptosporidium prevalence of `r round((Crypto_Detection %>% filter(Ct_mean > 0) %>% count() / Crypto_Detection %>% filter(Ct_mean >= 0) %>% count()) * 100, digits = 2)`% was detected in the House Mouse Hybrid Zone.

``` {r Pos-Samples-Yr, echo = F}
Pos_Samples_Yr <- Crypto_Detection %>% filter(Ct_mean > 0) %>% group_by(Year) %>% count()
Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv")

All_Samples_Yr <- Crypto_Detection %>% group_by(Year) %>% count()
Samples_Yr <- full_join(Pos_Samples_Yr, All_Samples_Yr, by = "Year") %>% mutate(Prevalence = (n.x / n.y)) %>% filter(!is.na(Prevalence))


Samples_Yr %>%
  ggplot(aes(x = Year, label = Prevalence)) +
      geom_col(aes(y = n.y, fill = "blue")) +
      geom_col(aes(y = n.x, fill = "red")) +
      geom_text(aes(label = percent(Prevalence),
                    y = (n.x / n.y)), nudge_y = 9) +
      labs(y = "Samples [n]") +
  theme(legend.position = "none") +
  theme(plot.title = element_text("Cryptosporidium Prevalence since 2016")) + 
  theme(plot.caption = element_text("Number of Samples caught per year (red) vs. number of Crypto-positive Samples per year, prevalence in percent"))
```

## Table of Highly Infected Samples
``` {r Top-infected, echo = F}
Samples <- Crypto_Detection %>% select(Mouse_ID, Ct_mean, Oocyst_Predict, Year, Latitude, Longitude) %>% filter(Ct_mean > 0) %>% arrange(Ct_mean) %>% head(20)

kable(Samples, caption = "Table 1.1: Crypto-Positive Samples collected since 2016, arranged by Ct_mean, 20 highest infected samples displayed", align = c("cccccc"))
```

## Geographic distribution of positive Samples in the HMHZ

```{r mini map HI , echo = F, message = F, warning = F}
SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") %>% select(-X) %>% filter(!is.na(Latitude), !is.na(Longitude), !is.na(HI), !Mouse_ID %in% c("SK_2697", "SK_3173", "SK_905"))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X)

Crypto_Detection_tested <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Ct_mean >= 0, !is.na(Latitude), !is.na(Longitude))

Crypto_Positive <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Ct_mean > 0, !is.na(Latitude), !is.na(Longitude))


Crypto_Detection_21 <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X) %>% filter(Year == 2021, Ct_mean > 0, !is.na(Latitude), !is.na(Longitude))

High_Infection_Samples <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_input/Cryptosporidium/High_Infection_Samples.csv") %>%
  mutate(HI_Indicator = case_when(HI >= 0.5 ~ "East",
                                  HI < 0.5 ~ "West"))

Western_Hybrid <- High_Infection_Samples %>% filter(HI >= 0.5)
Eastern_Hybrid <- High_Infection_Samples %>% filter(HI < 0.5)


map <- Crypto_Detection_21 %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 52.520007, lng =13.404954, zoom = 7)

# Cutting
SOTA$HI <- as.numeric(SOTA$HI)
SOTA$HI_Level <-  cut(SOTA$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$Ct_mean, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))

SeqSamples <- Crypto_Detection %>%
  mutate(Sequenced = Mouse_ID %in% c("AA_0144", "AA_0325", "AA_0689", "AA_0209",
                                     "AA_0282", "AA_0793", "AA_0667", "AA_0805", 
                                     "AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585",
                                     "AA_0546", "AA_0589", "AA_0571", "AA_0667"),
         HI_Indicator = case_when(HI >= 0.5 ~ "East",
                                  HI < 0.5 ~ "West"))


Ct_bl_30   <- Crypto_Detection_21 %>% filter(Ct_mean <= 30 & Ct_mean > 0)
Ct_bl_34   <- Crypto_Detection_21 %>% filter(Ct_mean <= 34 & Ct_mean > 30)
Ct_bl_36   <- Crypto_Detection_21 %>% filter(Ct_mean <= 36 & Ct_mean > 34)
Ct_over_36 <- Crypto_Detection_21 %>% filter(Ct_mean > 36)
Sequenced  <- SeqSamples %>% filter(Sequenced == T)
Not_Sequenced <- SeqSamples %>% filter(Sequenced == F)

data_col_HI       = colorFactor(beach(6), SOTA$HI)
data_col_HI_Level = colorFactor(beach(6), SOTA$HI_Level)
data_col_Ct_Level = colorFactor(brewer.pal(4, "YlOrRd"), Crypto_Detection_21$Ct_Level, reverse = T)
#data_col_Sequenced = colorFactor(brewer.pal(2, "YlOrRd"), SeqSamples$HI_Indicator, reverse = T)
data_col_Sequenced = colorFactor(beach(2), SeqSamples$HI)


map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875  , 52.6053, 51.8978, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 13.5000), color = "purple", weight = 55, opacity = 0.1) %>%
  addCircleMarkers(data = SOTA,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (total)") %>%
  addCircleMarkers(data = Crypto_Detection_tested,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (Crypto-tested)") %>%
  addCircleMarkers(data = Crypto_Positive,
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 1,
                   radius = 3,
                   group = "Samples (Crypto-positive)") %>%
  addCircleMarkers(data = Sequenced, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Sequenced") %>%
  addCircleMarkers(data = High_Infection_Samples, 
                   color = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Year:<b>",    as.character(Year),"<br>",
                                  "<b>Ct:<b>",      as.character(Ct_mean),"<br>",
                                  "<b>Oocysts:<b>", as.character(Oocyst_Predict), "<br>",
                                  "<b>Sex:<b>", Sex, "<br>",
                                  sep=" "),
                   opacity = 3,
                   radius = 3,
                   group = "High_Infection_Samples") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = SOTA$HI_Level, 
            group = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'),
            opacity = 1) %>%
addLayersControl(baseGroups = c("Samples (total)", "Samples (Crypto-tested)", "Samples (Crypto-positive)", "High_Infection_Samples", "Sequenced"), 
                 options = layersControlOptions(collapsed = F))
```

# Genes of Interest

## Genes and Gene_IDs
``` {r Gene-IDs, echo = F, message = F, warning = F}
Gene_IDs <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Gene_IDs.csv")

kable(Gene_IDs, align = c("lccccc"))
```

### GP60:
"Industry Standard" for Species- and Supspeciestyping in _Cryptosporidium spp._ 

- important in sporozoite motility and host invasion
- in _C.tyzzeri_ 5-8 TCA-repeats (encode for amino acid Serine)
- 8 repeats seen in 3 Czech samples
- 6 repeats in all other samples from Czech Republic and Germany

- additional 12bp repeat: (Consensus GGTACTCAAGGA)
  - 2 copies in IXa
  - 2 OR 3 copies in IXb??
  
- additional 18bp repeat
  - 2 copies in IXa (Consensus ATTCTGGTACTGAAGATA)
  - 2/3/4 copies in IXb (consensus GGTACTGAAAATAATTCT)


### TRAP-C1:
"Thrombospondin-related adhesive Protein C1"

- gliding motility
- host cell attachment
- host cell invasion

### MSC6-7:

- Serine-repeat antigen
- relatively high diversity in comparison to _C.parvum_ (38 SNPs, 54 Indels)
- highly polymorphic in other species
- potentially involved in host-adaptation


### GST:
"Glutathion-S-Transferase"

### MEDLE Multigene Family:
named after aa-motif at the end of the genes

- Crypto-specific gene family
- gene duplication
- potential role in host specificity

### SKSR Multigene family:

- similar to MEDLE family



# Sequencing Results from Kvac study and EH study

``` {r Map Prep, echo = F, message = F, warning = F}
SOTA <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/SOTA_Data_Product.csv") %>% select(-X) %>% filter(!is.na(Latitude), !is.na(Longitude), !is.na(HI), !Mouse_ID %in% c("SK_2697", "SK_3173", "SK_905"))

Crypto_Detection <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_products/Crypto_Detection.csv") %>% select(-X)

High_Infection_Samples <- read.csv("https://raw.githubusercontent.com/derele/Mouse_Eimeria_Field/master/data_input/Cryptosporidium/High_Infection_Samples.csv") %>%
  mutate(HI_Indicator = case_when(HI >= 0.5 ~ "East",
                                  HI < 0.5 ~ "West"))

Kvac <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Kvac_Samples_Locations1.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude)) 

colnames(Kvac)[colnames(Kvac)%in%"TRAP.C1_Subtype"] <- "TRAP_C1_Subtype"


Western_Hybrid <- High_Infection_Samples %>% filter(HI >= 0.5)
Eastern_Hybrid <- High_Infection_Samples %>% filter(HI < 0.5)



map <- Kvac %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 50.07245, lng =12.92014, zoom = 6)

# Cutting
SOTA$HI <- as.numeric(SOTA$HI)
SOTA$HI_Level <-  cut(SOTA$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'))
Crypto_Detection_21$Ct_Level <- cut(Crypto_Detection_21$Ct_mean, c(1, 28, 30, 32, 34, 36, 40), include.lowest = F , labels = c(' < 28  ', '28 - 30', '30 - 32', '32 - 34', '34 - 36', ' > 36'))



GP60     <- Kvac %>% filter(!is.na(GP60_Subtype))
Actin    <- Kvac %>% filter(Actin_Subtype %in% c("A1", "A2"))
TRAP_C1  <- Kvac %>% filter(TRAP_C1_Subtype %in% c("T1", "T2"))
COWP     <- Kvac %>% filter(COWP_Subtype %in% c("C1", "C2"))


data_col_HI    = colorFactor(beach(6), Kvac$HI)
data_col_GP60  = colorFactor(beach(9), GP60$GP60_Subtype, rev = T)
data_col_Actin = colorFactor(beach(2), Actin$Actin_Subtype, rev = T)
data_col_Trap  = colorFactor(beach(2), TRAP_C1$TRAP_C1_Subtype, rev = T)
data_col_COWP  = colorFactor(beach(2), COWP$COWP_Subtype, rev = T)


SeqSamples <- Crypto_Detection %>%
  mutate(Sequenced = Mouse_ID %in% c("AA_0144", "AA_0325", "AA_0689", "AA_0209",
                                     "AA_0282", "AA_0793", "AA_0667", "AA_0805", 
                                     "AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585",
                                     "AA_0546", "AA_0589", "AA_0571"),
         HI_Indicator = case_when(HI >= 0.5 ~ "East",
                                  HI < 0.5 ~ "West"),
         TRAP_C1_Subtype = case_when(Mouse_ID %in% c("AA_0546", "AA_0578", "AA_0580", "AA_0589") ~ "Eastern",
                                     Mouse_ID %in% c("AA_0571", "AA_0667") ~ "Western"),
         GP60_Subtype = case_when(Mouse_ID %in% c("AA_0900", "AA_0523",
                                     "AA_0534", "AA_0537", "AA_0545", "AA_0553",
                                     "AA_0554", "AA_0578", "AA_0580", "AA_0585") ~ "IXb"),
         MSC6_Subtype = case_when(Mouse_ID %in% c("AA_0282", "AA_0523") ~ "Substitution T->A @236nt and C->T @350",
                                  Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0667", "AA_0689", "AA_0793", "AA_0805", "AA_0900") ~ "A @236nt and T @350"),
         GST_Subtype = case_when(Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "A->T @519",
                         Mouse_ID %in% c("AA_0523") ~ "like RefSeq"),
         SKSR_Subtype = case_when(Mouse_ID %in% c("AA_0523") ~ "more SNPs",
                                  Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "different from AA_0523"),
         MEDLE_Subtype = case_when(Mouse_ID %in% c("AA_0523") ~ "Deletion of 9nt @446",
                                   Mouse_ID %in% c("AA_0144", "AA_0209", "AA_0900") ~ "Deletion of 6nt @446"))

```


## Kvac Study Results
``` {r Kvac-Samples-Table, echo = F, message = F, fig.align = 'left'}
HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Kvac_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))

Kvac <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Kvac_Samples_Locations1.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude)) 

colnames(Kvac)[colnames(Kvac)%in%"TRAP.C1_Subtype"] <- "TRAP_C1_Subtype"

Kvac <- left_join(Kvac, HMHZ)  %>% select(1, 7:11)  %>% mutate(GP60_Subtype = case_when(!is.na(GP60) ~ "seq")) %>% select(-6)

Kvac <- Kvac %>% 
  pivot_longer(cols = c(2:5), names_to = "Marker", values_to = "Sequenced1") %>% 
  mutate(Sequenced = case_when(!is.na(Sequenced1) ~ TRUE,
                               is.na(Sequenced1)  ~ FALSE)) %>% 
  select(-Sequenced1)

Kvac %>% 
  ggplot(aes(x = Marker, y = Mouse_ID, fill = Sequenced)) +
  geom_tile() +
  scale_fill_paletteer_d("beyonce::X2") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```

## HMHZ Sequencing Reults
``` {r HMHZ-Samples-Table, echo = F, message = F, fig.align = 'left'}
HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Kvac_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))

HMHZ <- HMHZ[HMHZ$Mouse_ID %like% "AA_", ]

HMHZ[,c(1:6, 8:17)] %>% kable(align = c("ccccccccccccccccc"))

```


## Map of Sequencing Results
```{r mini map Sequencing-Results, echo = F, message = F, warning = F}

HMHZ <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Kvac_Samples_Locations.csv", na.strings=c(""," ","NA")) %>% filter(!is.na(Longitude))

Sequenced <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Samples_Locations.csv")

Sequenced$HI_Level <-  cut(Sequenced$HI, c(0, 0.001, 0.250, 0.500, 0.750, 0.999, 1), include.lowest = T , labels = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'))

Sequenced <- Sequenced %>%
  mutate(HI_Indicator = case_when(MSC6_Subtype == "MS1" ~ "Eastern",
                                  MSC6_Subtype == "MS2" ~ "Western"))

Sequenced <- full_join(Sequenced, Kvac)
Sequenced <- full_join(Sequenced, HMHZ) %>% arrange(Mouse_ID) %>% group_by(Mouse_ID) %>% fill(c(everything()), .direction = "downup") %>% ungroup() %>% distinct(Mouse_ID, .keep_all = T) 


colnames(Sequenced)[colnames(Sequenced)%in%"TRAP.C1_Subtype"] <- "TRAP_C1_Subtype"




Ct_bl_30   <- Crypto_Detection_21 %>% filter(Ct_mean <= 30 & Ct_mean > 0)
Ct_bl_34   <- Crypto_Detection_21 %>% filter(Ct_mean <= 34 & Ct_mean > 30)
Ct_bl_36   <- Crypto_Detection_21 %>% filter(Ct_mean <= 36 & Ct_mean > 34)
Ct_over_36 <- Crypto_Detection_21 %>% filter(Ct_mean > 36)
GST   <- Sequenced %>% filter(!is.na(GST_Subtype))
MSC6  <- Sequenced %>% filter(!is.na(MSC6_Subtype))
GP60  <- Sequenced %>% filter(!is.na(GP60_Subtype))
TRAP  <- Sequenced %>% filter(!is.na(TRAP_C1_Subtype))
SKSR  <- Sequenced %>% filter(!is.na(SKSR_Subtype))
MEDLE <- Sequenced %>% filter(!is.na(MEDLE_Subtype))
CP56  <- Sequenced %>% filter(!is.na(CP56_Subtype))
GP60_Prot <- Sequenced %>% filter((!is.na(GP60_Subtype_Protein)))

data_col_HI           = colorFactor(beach(6), Sequenced$HI)
data_col_HI_Level     = colorFactor(beach(6), Sequenced$HI_Level)
data_col_HI_Indicator = colorFactor(beach(2), Sequenced$HI_Indicator, reverse = T)
data_col_Sequenced    = colorFactor(beach(2), Sequenced$HI)
data_col_MSC6         = colorFactor(beach(2), Sequenced$MSC6_Subtype, rev = T)
data_col_GP60         = colorFactor(beach(2), Sequenced$GP60_Subtype, rev = T)
data_col_GP60_Prot    = colorFactor(beach(7), Sequenced$GP60_Subtype_Protein, reverse = T)
data_col_GST   = colorFactor(beach(2), Sequenced$GST_Subtype, rev = T)
data_col_Trap  = colorFactor(beach(2), Sequenced$TRAP_C1_Subtype, rev = T)
data_col_SKSR  = colorFactor(beach(2), Sequenced$SKSR_Subtype, rev = T)
data_col_MEDLE = colorFactor(beach(2), Sequenced$MEDLE_Subtype, rev = T)
data_col_CP56  = colorFactor(beach(2), Sequenced$CP56_Subtype, rev = T)



map <- Sequenced %>%
  leaflet() %>%
  addProviderTiles("CartoDB") %>%
  setView(lat = 51.676761 , lng =13.679507, zoom = 6)

map %>%
  addPolylines(lat = c(55.0000, 53.6000, 53.51885, 52.8875  , 52.6053, 51.8978, 45.0000), lng = c(10.0000, 11.4563, 12.4464,13.8119 , 13.8756, 13.8103, 13.5000), color = "purple", weight = 55, opacity = 0.1) %>%
  addCircleMarkers(data = Sequenced, 
                   col = ~data_col_HI(HI),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by HI") %>%
    addCircleMarkers(data = MSC6, 
                   col = ~data_col_MSC6(MSC6_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>", 
                                  "<b>GP60 Subtype:<b>",        as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by MSC6-7") %>%
  
  addCircleMarkers(data = GST, 
                   col = ~data_col_GST(GST_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GST") %>%
  addCircleMarkers(data = SKSR, 
                   col = ~data_col_SKSR(SKSR_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by SKSR") %>%
  addCircleMarkers(data = MEDLE, 
                   col = ~data_col_MEDLE(MEDLE_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by MEDLE") %>%
  addCircleMarkers(data = CP56, 
                   col = "red",
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by CP56") %>%
  addCircleMarkers(data  = GP60, 
                   col   = ~data_col_GP60(GP60_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  "<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GP60") %>%
  addCircleMarkers(data  = GP60_Prot, 
                   col   = ~data_col_GP60_Prot(GP60_Subtype_Protein),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  "<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3,
                   group = "Colored by GP60 Detail") %>%
  addCircleMarkers(data = TRAP, 
                   col = ~data_col_Trap(TRAP_C1_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude), "<br>",
                                  "<b>GP60 Subtype:<b>",      as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>MSC6-7 Subtype:<b>",      as.character(MSC6_Subtype), "<br>",
                                  "<b>GST Subtype:<b>",      as.character(GST_Subtype), "<br>",
                                  "<b>CP56 Subtype:<b>",      as.character(CP56_Subtype), "<br>",
                                  "<b>SKSR Subtype:<b>",      as.character(SKSR_Subtype), "<br>",
                                  "<b>MEDLE Subtype:<b>",      as.character(MEDLE_Subtype), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius = 3, 
                   group = "Colored by TRAP-C1") %>%
    addCircleMarkers(data  = Actin, 
                   col   = ~data_col_Actin(Actin_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude),  "<br>",
                                  "<b>GP60 Subtype:<b>",       as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  
                                  sep=" "),
                   opacity = 5,
                   radius  = 3,
                   group   = "Colored by Actin") %>%
  addCircleMarkers(data  = COWP, 
                   col   = ~data_col_COWP(COWP_Subtype),
                   label = ~htmlEscape(Mouse_ID),
                   popup = ~paste("<b>Mouse_ID:<b>",as.character(Mouse_ID), "<br>",
                                  "<b>HI:<b>",      as.character(HI), "<br>",
                                  "<b>Location:<b>", as.character(Latitude), "<b>,<b>", as.character(Longitude),
                                  "<b>GP60 Subtype:<b>",       as.character(GP60_Subtype), "<br>",
                                  #"<b>GP60 Detail Subtype:<b>", as.character(GP60_Subtype_Protein), "<br>",
                                  "<b>Actin Subtype:<b>",      as.character(Actin_Subtype), "<br>",
                                  "<b>TRAP-C1 Subtype:<b>",    as.character(TRAP_C1_Subtype), "<br>",
                                  "<b>COWP Subtype:<b>",      as.character(COWP_Subtype), "<br>",
                                  sep=" "),
                   opacity = 5,
                   radius  = 3,
                   group   = "Colored by COWP") %>%
  addLegend("bottomleft", 
            pal = data_col_HI_Level, 
            title = "HI",
            values = Sequenced$HI_Level, 
            group = c('HI = 0', 'HI < 0.25', 'HI < 0.5', 'HI < 0.75', 'HI < 1', 'HI = 1'),
            opacity = 1) %>%
  addLegend("bottomright", 
            pal = data_col_HI_Indicator, 
            title = "Type",
            values = Sequenced$HI_Indicator, 
            group = c('Western', 'Eastern'),
            opacity = 1) %>%
addLayersControl(baseGroups = c("Colored by HI", 
                                "Colored by MSC6-7", 
                                "Colored by GST",
                                "Colored by SKSR",
                                "Colored by MEDLE",
                                "Colored by CP56",
                                "Colored by GP60",
                                "Colored by GP60 Detail",
                                "Colored by TRAP-C1",
                                "Colored by COWP",
                                "Colored by Actin"),
                 options = layersControlOptions(collapsed = F))
```



# Consensus Trees and Heatmap

## Sequenced Samples

Concatenating helps understanding HOW different individual samples are, but since I don't have sequences for all my genes (gp60, MSC6-7, SKSR, MEDLE, GST, and CP56) from all Samples (AA_0144 - AA_0900), I have to cut down the number of sequences (less genes in the comparison) or the number of samples (less samples, but more sequences).

``` {r Logical_Sequenced, message = F, warning=FALSE, echo=F}
Seq_Logical <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Seq_Logical.csv") %>% mutate(GP60 = TRUE) %>% select(c(1, 6:12)) %>% pivot_longer(cols = c(2:8), names_to = "Marker", values_to = "Sequenced")

Seq_Logical %>%
  ggplot(aes(x = Marker, y = Mouse_ID, fill = Sequenced)) +
  geom_tile() +
  scale_fill_paletteer_d("beyonce::X2") +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5))

```


### Tree #1: less markers (GP60 and MSC6-7) for 9 samples:
``` {r Limited Genes Distance Matrix, message = F, warning=FALSE, echo=F}

Genes_Limited <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Limited.csv")

Genes_Limited <- Genes_Limited[, 2:10]
Genes_Limited_Matrix <- as.matrix(Genes_Limited)
pheatmap(Genes_Limited_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

```

![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Limited_Genes.png)


### Tree #2: more markers (gp60, MSC6-7, SKSR, MEDLE, GST) for 4 samples

Concatenated Sequences of the following genes:

- GP60
- MSC6-7
- SKSR
- MEDLE
- GST
- (CP56 is only available for 2/4 samples, so it was not used here)

Samples of Interest:

- AA_0144 (CP56 available, but dropped)
- AA_0209 (CP56 available, but dropped)
- AA_0523
- AA_0900

``` {r Max Genes Distance Matrix, message = F, warning=FALSE, echo=F}
Genes_Max <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Max.csv")

Genes_Max <- Genes_Max[, 2:5]
Genes_Max_Matrix <- as.matrix(Genes_Max)
pheatmap(Genes_Max_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

#![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Max_Genes.png)


```

## Heatmap of Kvac and HMHZ Samples for GP60

GP60 Alignment

![Alignment of all available gp60 alignments from Kvac Study and EH-HMHZ samples](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Alignment.png)




```{r Distance Matrix, message = F, warning=FALSE, echo=F}
GP60_Distance <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree.csv")
GP60_Distance <- GP60_Distance[-c(36:40), -c(37:41)]


#dist_GP60 <- dist(GP60_Distance, method = "euclidean")
#hc_gp60   <- hclust(dist_GP60, method = "average")
#dend_gp60 <- as.dendrogram(hc_gp60)
#  plot(dend_gp60)
#dend_colored_gp60 <- color_branches(dend_gp60, h = 0.1)
#  plot(dend_colored_gp60)


GP60_Distance <- GP60_Distance[, 2:46]
#pheatmap(as.matrix(GP60_Distance), 
#        scale = "column", 
#        col = heat.colors(256), 
#        main = "GP60 subtype families",
#        Rowv = NA,
#        Colv = NA, cluster_rows = T)

GP60_Distance_matrix <- as.matrix(GP60_Distance)
pheatmap(GP60_Distance_matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "GP60 Heatmap of HMHZ Samples based on Protein Sequences")

```

Geneious Tree for the GP60 Sequences from Kvac and EH Study

``` {r Geneious-Tree-3, echo = F, fig.cap = "Geneious Tree of available GP60 Sequences from Kvac study, EH study. Cost matrix of 65% Similarity (5.0/-4.0), Gap open penalty 12, Gap extension penalty 3, Global Alignment, Genetic Distance Model Tamura-Nei, Tree build method Neighbor-Joining, no Outgroup"}

knitr::include_graphics("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Protein_Tree_GP60.png")

```


# Trees with PhyloTree

``` {r Phylogenetic Trees in R, echo = F, message = F, warning = F}

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")
BiocManager::install("phyloTree")

library(ggtree)
library(phyloTree)

```