---
title: "Trash I guess"
output: html_document
---


# Consensus Trees and Heatmap

## Sequenced Samples

Concatenating helps understanding HOW different individual samples are, but since I don't have sequences for all my genes (gp60, MSC6-7, SKSR, MEDLE, GST, and CP56) from all Samples (AA_0144 - AA_0900), I have to cut down the number of sequences (less genes in the comparison) or the number of samples (less samples, but more sequences).

``` {r Logical_Sequenced, message = F, warning=FALSE, echo=F}
Seq_Logical <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Seq_Logical.csv") %>% mutate(GP60 = TRUE) %>% select(c(1, 6:12)) %>% pivot_longer(cols = c(2:8), names_to = "Marker", values_to = "Sequenced")

Seq_Logical %>%
  ggplot(aes(x = Marker, y = Mouse_ID, fill = Sequenced)) +
  geom_tile() +
  scale_fill_paletteer_d("beyonce::X2") +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5))

```


### Tree #1: less markers (GP60 and MSC6-7) for 9 samples:
``` {r Limited Genes Distance Matrix, message = F, warning=FALSE, echo=F}

Genes_Limited <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Limited.csv")

Genes_Limited <- Genes_Limited[, 2:10]
Genes_Limited_Matrix <- as.matrix(Genes_Limited)
pheatmap(Genes_Limited_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

```

![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Limited_Genes.png)


### Tree #2: more markers (gp60, MSC6-7, SKSR, MEDLE, GST) for 4 samples

Concatenated Sequences of the following genes:

- GP60
- MSC6-7
- SKSR
- MEDLE
- GST
- (CP56 is only available for 2/4 samples, so it was not used here)

Samples of Interest:

- AA_0144 (CP56 available, but dropped)
- AA_0209 (CP56 available, but dropped)
- AA_0523
- AA_0900

``` {r Max Genes Distance Matrix, message = F, warning=FALSE, echo=F}
Genes_Max <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Max.csv")

Genes_Max <- Genes_Max[, 2:5]
Genes_Max_Matrix <- as.matrix(Genes_Max)
pheatmap(Genes_Max_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

#![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Max_Genes.png)


```

## Heatmap of Kvac and HMHZ Samples for GP60

GP60 Alignment

![Alignment of all available gp60 alignments from Kvac Study and EH-HMHZ samples](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Alignment.png)




```{r Distance Matrix, message = F, warning=FALSE, echo=F}
GP60_Distance <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree.csv")
GP60_Distance <- GP60_Distance[-c(36:40), -c(37:41)]


#dist_GP60 <- dist(GP60_Distance, method = "euclidean")
#hc_gp60   <- hclust(dist_GP60, method = "average")
#dend_gp60 <- as.dendrogram(hc_gp60)
#  plot(dend_gp60)
#dend_colored_gp60 <- color_branches(dend_gp60, h = 0.1)
#  plot(dend_colored_gp60)


GP60_Distance <- GP60_Distance[, 2:46]
#pheatmap(as.matrix(GP60_Distance), 
#        scale = "column", 
#        col = heat.colors(256), 
#        main = "GP60 subtype families",
#        Rowv = NA,
#        Colv = NA, cluster_rows = T)

GP60_Distance_matrix <- as.matrix(GP60_Distance)
pheatmap(GP60_Distance_matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "GP60 Heatmap of HMHZ Samples based on Protein Sequences")

```

Geneious Tree for the GP60 Sequences from Kvac and EH Study

``` {r Geneious-Tree-3, echo = F, fig.cap = "Geneious Tree of available GP60 Sequences from Kvac study, EH study. Cost matrix of 65% Similarity (5.0/-4.0), Gap open penalty 12, Gap extension penalty 3, Global Alignment, Genetic Distance Model Tamura-Nei, Tree build method Neighbor-Joining, no Outgroup"}

knitr::include_graphics("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Protein_Tree_GP60.png")

```



### GP60:
"Industry Standard" for Species- and Supspeciestyping in _Cryptosporidium spp._ 

- important in sporozoite motility and host invasion
- in _C.tyzzeri_ 5-8 TCA-repeats (encode for amino acid Serine)
- 8 repeats seen in 3 Czech samples
- 6 repeats in all other samples from Czech Republic and Germany

- additional 12bp repeat: (Consensus GGTACTCAAGGA)
  - 2 copies in IXa
  - 2 OR 3 copies in IXb??
  
- additional 18bp repeat
  - 2 copies in IXa (Consensus ATTCTGGTACTGAAGATA)
  - 2/3/4 copies in IXb (consensus GGTACTGAAAATAATTCT)


### TRAP-C1:
"Thrombospondin-related adhesive Protein C1"

- gliding motility
- host cell attachment
- host cell invasion


### COWP:
"Oocyst Wall Protein"

- essential in Sexual Reproduction of Parasite

### MSC6-7:

- Serine-repeat antigen
- relatively high diversity in comparison to _C.parvum_ (38 SNPs, 54 Indels)
- highly polymorphic in other species
- potentially involved in host-adaptation


### GST:
"Glutathion-S-Transferase"

### MEDLE Multigene Family:
named after aa-motif at the end of the genes

- Crypto-specific gene family
- gene duplication
- potential role in host specificity

### SKSR Multigene family:

- similar to MEDLE family








Concatenating helps understanding HOW different individual samples are, but since I don't have sequences for all my genes (gp60, MSC6-7, SKSR, MEDLE, GST, and CP56) from all Samples (AA_0144 - AA_0900), I have to cut down the number of sequences (less genes in the comparison) or the number of samples (less samples, but more sequences).

``` {r Logical_Sequenced, message = F, warning=FALSE, echo=F}
Seq_Logical <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/HMHZ_Seq_Logical.csv") %>% mutate(GP60 = TRUE) %>% select(c(1, 6:12)) %>% pivot_longer(cols = c(2:8), names_to = "Marker", values_to = "Sequenced")

Seq_Logical %>%
  ggplot(aes(x = Marker, y = Mouse_ID, fill = Sequenced)) +
  geom_tile() +
  scale_fill_paletteer_d("beyonce::X2") +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5))

```


### Tree #1: less markers (GP60 and MSC6-7) for 9 samples:
``` {r Limited Genes Distance Matrix, message = F, warning=FALSE, echo=F}

Genes_Limited <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Limited.csv")

Genes_Limited <- Genes_Limited[, 2:10]
Genes_Limited_Matrix <- as.matrix(Genes_Limited)
pheatmap(Genes_Limited_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

```

![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Limited_Genes.png)


### Tree #2: more markers (gp60, MSC6-7, SKSR, MEDLE, GST) for 4 samples

Concatenated Sequences of the following genes:

- GP60
- MSC6-7
- SKSR
- MEDLE
- GST
- (CP56 is only available for 2/4 samples, so it was not used here)

Samples of Interest:

- AA_0144 (CP56 available, but dropped)
- AA_0209 (CP56 available, but dropped)
- AA_0523
- AA_0900

``` {r Max Genes Distance Matrix, message = F, warning=FALSE, echo=F}
Genes_Max <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree_Max.csv")

Genes_Max <- Genes_Max[, 2:5]
Genes_Max_Matrix <- as.matrix(Genes_Max)
pheatmap(Genes_Max_Matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "Heatmap of HMHZ Genes GP60 and MSC6-7", cluster_cols = T)

#![](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Max_Genes.png)


```

## Heatmap of Kvac and HMHZ Samples for GP60

GP60 Alignment

![Alignment of all available gp60 alignments from Kvac Study and EH-HMHZ samples](https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Alignment.png)




```{r Distance Matrix, message = F, warning=FALSE, echo=F}
GP60_Distance <- read.csv("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/GP60_Prot_Tree.csv")
GP60_Distance <- GP60_Distance[-c(36:40), -c(37:41)]


#dist_GP60 <- dist(GP60_Distance, method = "euclidean")
#hc_gp60   <- hclust(dist_GP60, method = "average")
#dend_gp60 <- as.dendrogram(hc_gp60)
#  plot(dend_gp60)
#dend_colored_gp60 <- color_branches(dend_gp60, h = 0.1)
#  plot(dend_colored_gp60)


GP60_Distance <- GP60_Distance[, 2:46]
#pheatmap(as.matrix(GP60_Distance), 
#        scale = "column", 
#        col = heat.colors(256), 
#        main = "GP60 subtype families",
#        Rowv = NA,
#        Colv = NA, cluster_rows = T)

GP60_Distance_matrix <- as.matrix(GP60_Distance)
pheatmap(GP60_Distance_matrix, show_rownames = F, border_color = NA, cluster_rows = T, main = "GP60 Heatmap of HMHZ Samples based on Protein Sequences")

```

Geneious Tree for the GP60 Sequences from Kvac and EH Study

``` {r Geneious-Tree-3, echo = F, fig.cap = "Geneious Tree of available GP60 Sequences from Kvac study, EH study. Cost matrix of 65% Similarity (5.0/-4.0), Gap open penalty 12, Gap extension penalty 3, Global Alignment, Genetic Distance Model Tamura-Nei, Tree build method Neighbor-Joining, no Outgroup"}

knitr::include_graphics("https://raw.githubusercontent.com/tlobnow/Cryptosporidium-BSc/Main-Branch/Analysis/Protein_Tree_GP60.png")

```


# Trees with PhyloTree

``` {r Phylogenetic Trees in R, echo = F, message = F, warning = F}

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")
BiocManager::install("phyloTree")

library(ggtree)
library(phyloTree)

```